cpp/include/graph.hpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+namespace cugraph {
+namespace experimental {
+
+typedef enum prop_type{PROP_UNDEF, PROP_FALSE, PROP_TRUE} PropType;
+
+struct GraphProperties {
+  bool directed;
+  bool weighted;
+  bool multigraph;
+  bool bipartite;
+  bool tree;
+  PropType has_negative_edges;
```

Comment:
bool directed{false};
bool weighted{false};
bool multigraph{false};
bool tree{false};
PropType has_negative_edges{prop_type::PROP_UNDEF};

will be simpler & clearer than adding a constructor for simply value initialization.